test/common/upstream/cds_api_impl_test.cc

```diff
@@ -77,12 +77,17 @@ class CdsApiImplTest : public testing::Test {
         .WillOnce(Invoke(
             [&](Http::MessagePtr& request, Http::AsyncClient::Callbacks& callbacks,
                 const absl::optional<std::chrono::milliseconds>&) -> Http::AsyncClient::Request* {
-              EXPECT_EQ((Http::TestHeaderMapImpl{
-                            {":method", v2_rest_ ? "POST" : "GET"},
-                            {":path", v2_rest_ ? "/v2/discovery:clusters"
-                                               : "/v1/clusters/cluster_name/node_name"},
-                            {":authority", "foo_cluster"}}),
-                        request->headers());
+              Http::TestHeaderMapImpl expected_headers{
+                  {":method", v2_rest_ ? "POST" : "GET"},
```

Comment:
Any reason to not just do this for v1 also?

Reply:
I think it's probably work doing in both places since it's more correct and it will remove the special cases from the tests?

Reply:
Sorry, a clarification: If I get you correctly, you want to make v1 SDS sends requests using `POST` (and send along node information to the management server) as well? Since the current SDS request only sets `method` (as `GET`) and `path` only.

https://github.com/envoyproxy/envoy/blob/2662bf1f220a0c86289c8ead6a85cab388e8c14a/source/common/upstream/sds_subscription.cc#L86-L87

Or keep it `GET` but send it via headers (but this is a different story).

Reply:
I was just suggesting to set content-length and content-type for the v1 requests so the tests don't need special casing?

Reply:
I don't think we can do that since the `GET` request for v1 has no payload sent along with the request.